### ISSUE TITLE
Disabled listbox items (e.g. in Dropdown) should still be focusable

### ISSUE THREAD

**Issue by craigkovatch** [microsoft/fluentui#17205]
### Environment Information

- **Package version(s)**: latest (7?)
- **Browser and OS versions**: Chrome 88 on macOS

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/craigkovatch/pen/rNWdVMv

#### Actual behavior:

Disabled items in the popup listbox of the Dropdown control are not keyboard-navigable.

#### Expected behavior:

Disabled items in the popup listbox should remain keyboard-navigable, as they are on e.g. the ComboBox popup.

### Documentation describing expected behavior

Aria Practices Guide suggests that disabled listbox options should remain focusable: https://www.w3.org/TR/wai-aria-practices-1.2/#kbd_disabled_controls

**Comment by joschect** [comment]
@craigkovatch Thanks for the submission! I believe this was done to keep the dropdown component inline with the way that default select works. You can see [in this codepen](https://codepen.io/slimemold/pen/ZEBoJQE) that the regular select skips disabled items as well. I'm not positive but I believe that aria 1.2 is still under development as per [this url](https://www.w3.org/WAI/standards-guidelines/aria/).

@smhigley Might be able to say more.

**Comment by craigkovatch** [comment]
Ah the recommendation is not new in the 1.2 Practices guide, same in 1.1: https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_disabled_controls

I was confused that the disabled options were focusable on the ComboBox popup but not the Dropdown popups. If this is an intentional difference between the two paradigms, I would love to understand the reasoning, for my own learning!

Thanks :)

**Comment by smhigley** [comment]
ARIA 1.2 is a candidate rec as of today! 😄

I'd lean towards keeping the disabled options focusable within all combobox/dropdown/picker-type components, since otherwise they wouldn't be discoverable at all -- virtual cursor isn't really used and often doesn't work within listboxes. It's also a good point that we should have a consistent approach between listboxes in different components.

**Comment by smhigley** [comment]
@khmakoto (and generally @microsoft/fluentui-react-pickers-and-more) -- would there be any concerns with changing this behavior in v8?

**Comment by khmakoto** [comment]
My 2 cents here, I think making a change like this that only affects aria + focus and creates consistency and improvement overall is worth it.
